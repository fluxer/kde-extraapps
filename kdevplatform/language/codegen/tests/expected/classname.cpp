/*
 * This is just a test.
 * You may do with it as you please.
 */

#include "classname.h"

void ClassName::doSomething(double howMuch, bool doSomethingElse)
{
    
}

int ClassName::getSomeOtherNumber()
{
    
}
