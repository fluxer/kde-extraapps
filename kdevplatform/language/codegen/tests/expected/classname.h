/*
 * This is just a test.
 * You may do with it as you please.
 */

#ifndef CLASSNAME_H
#define CLASSNAME_H

class ClassName : public QObject
{
public:
    void doSomething(double howMuch, bool doSomethingElse);
    int getSomeOtherNumber();

    QString name;
    int number;
    SomeCustomType data;
};

#endif // CLASSNAME_H
