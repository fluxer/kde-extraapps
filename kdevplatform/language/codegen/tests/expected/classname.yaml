class:
  name: ClassName
  members:
  - name: name
    type: QString
  - name: number
    type: int
  - name: data
    type: SomeCustomType
  functions:
  - name: doSomething
    arguments:
    - name: howMuch
      type: double
    - name: doSomethingElse
      type: bool
  - name: getSomeOtherNumber
    return values:
    - name: someOtherNumber
      type: int
